config/v1/types_oauth.go

```diff
@@ -10,18 +10,22 @@ import (
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// OAuth holds cluster-wide information about OAuth.  The canonical name is `cluster`
+// OAuth holds cluster-wide information about OAuth.  The canonical name is `cluster`.
+// It is used to configure the integrated OAuth server.
+// This configuration is only honored when the top level Authentication config has type set to IntegratedOAuth.
 type OAuth struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
 	Spec              OAuthSpec   `json:"spec"`
-	Status            OAuthStatus `json:"status,omitempty"`
+	Status            OAuthStatus `json:"status"`
 }
 
 // OAuthSpec contains desired cluster auth configuration
 type OAuthSpec struct {
-	// identityProviders is an ordered list of ways for a user to identify themselves
-	IdentityProviders []OAuthIdentityProvider `json:"identityProviders"`
+	// identityProviders is an ordered list of ways for a user to identify themselves.
+	// When this list is empty, no identities are provisioned for users.
+	// +optional
+	IdentityProviders []IdentityProvider `json:"identityProviders"`
```

Comment:
If this is `+optional`, shouldn't it be `omitempty`?

Reply:
Sounds reasonable to me, though we seem to be inconsistent on this (I guess that is expected when the same-ish information is stored in multiple places).